app/scripts/snaps/preinstalled-snaps.ts

```diff
@@ -0,0 +1,57 @@
+import type { PreinstalledSnap } from '@metamask/snaps-controllers';
+
+// Preinstalled snaps requires fs and require to read and use any installed snaps.
+// We can switch `require` to `import`, but then this file gets transpiled to ESM & won't have access to `require.resolve`.
+// Work is needed to create the require object in ESM - specifically the import url. See snippet
+// import { createRequire } from 'node:module'
+// const require = createRequire(import.meta.url)
+//                                ^ The 'import.meta' meta-property is not allowed in files which will build into CommonJS output
+/* eslint-disable-next-line @typescript-eslint/no-require-imports,@typescript-eslint/no-var-requires */
+const fs = require('fs');
```

Comment:
Why are you overriding our eslint rules?

Reply:
See the comment above.

1. Importing `fs` or `node:fs` does not work due to (I believe) lavamoat.
2. Unfortunately I was unable to get the ESM `createRequire` syntax to work.

Options:
1. We convert this file to (C)JS so we can use the `require` syntax, but believe we are moving away from JS scripts.
2. We disable the linting rule.

Reply:
@davidmurdoch and or @legobeat do you have any feedback on this item?

just to note the fs that will be imported here is the browserify version of fs, not node. At least thats my understanding because this code is run in the browser which doesn't have fs.

Reply:
Oooh browserify, that must mean we are also using browserify's `require` here too.
If this linting rule is valid, then why did we not get any errors or warning when using `require.resolve`?

<details><summary>Summary when using  `import fs from 'fs'`</summary>

```
scripts:core:dev:standardEntryPoints: Error: tried to statically call { readFileSync: [Function: readFileSync], readFile: [Function: readFile], readdirSync: [Function: readdirSync], readdir: [Function: readdir] } as a function while parsing file: /Users/prithpalsooriya/Desktop/projects/metamask/metamask-extension/app/scripts/snaps/preinstalled-snaps.ts
    at error (/Users/prithpalsooriya/Desktop/projects/metamask/metamask-extension/node_modules/static-module/index.js:167:45)
    at traverse (/Users/prithpalsooriya/Desktop/projects/metamask/metamask-extension/node_modules/static-module/index.js:275:24)
    at walk (/Users/prithpalsooriya/Desktop/projects/metamask/metamask-extension/node_modules/static-module/index.js:259:13)
    at c (/Users/prithpalsooriya/Desktop/projects/metamask/metamask-extension/node_modules/acorn-walk/dist/walk.js:79:24)
    at Object.skipThrough (/Users/prithpalsooriya/Desktop/projects/metamask/metamask-extension/node_modules/acorn-walk/dist/walk.js:186:39)
    at c (/Users/prithpalsooriya/Desktop/projects/metamask/metamask-extension/node_modules/acorn-walk/dist/walk.js:78:24)
    at base.NewExpression.base.CallExpression (/Users/prithpalsooriya/Desktop/projects/metamask/metamask-extension/node_modules/acorn-walk/dist/walk.js:395:11)
    at c (/Users/prithpalsooriya/Desktop/projects/metamask/metamask-extension/node_modules/acorn-walk/dist/walk.js:78:24)
    at Object.skipThrough (/Users/prithpalsooriya/Desktop/projects/metamask/metamask-extension/node_modules/acorn-walk/dist/walk.js:186:39)
    at c (/Users/prithpalsooriya/Desktop/projects/metamask/metamask-extension/node_modules/acorn-walk/dist/walk.js:78:24)
    at base.VariableDeclarator (/Users/prithpalsooriya/Desktop/projects/metamask/metamask-extension/node_modules/acorn-walk/dist/walk.js:285:22)
    at c (/Users/prithpalsooriya/Desktop/projects/metamask/metamask-extension/node_modules/acorn-walk/dist/walk.js:78:24)
    at base.VariableDeclaration (/Users/prithpalsooriya/Desktop/projects/metamask/metamask-extension/node_modules/acorn-walk/dist/walk.js:280:7)
    at c (/Users/prithpalsooriya/Desktop/projects/metamask/metamask-extension/node_modules/acorn-walk/dist/walk.js:78:24)
    at Object.skipThrough (/Users/prithpalsooriya/Desktop/projects/metamask/metamask-extension/node_modules/acorn-walk/dist/walk.js:186:39)
    at c (/Users/prithpalsooriya/Desktop/projects/metamask/metamask-extension/node_modules/acorn-walk/dist/walk.js:78:24)
    at base.Program.base.BlockStatement (/Users/prithpalsooriya/Desktop/projects/metamask/metamask-extension/node_modules/acorn-walk/dist/walk.js:198:7)
    at c (/Users/prithpalsooriya/Desktop/projects/metamask/metamask-extension/node_modules/acorn-walk/dist/walk.js:78:24)
    at Object.full (/Users/prithpalsooriya/Desktop/projects/metamask/metamask-extension/node_modules/acorn-walk/dist/walk.js:80:7)
    at full (/Users/prithpalsooriya/Desktop/projects/metamask/metamask-extension/node_modules/acorn-node/walk.js:20:15)
    at /Users/prithpalsooriya/Desktop/projects/metamask/metamask-extension/node_modules/static-module/index.js:57:13
    at ConcatStream.<anonymous> (/Users/prithpalsooriya/Desktop/projects/metamask/metamask-extension/node_modules/concat-stream/index.js:37:43)
    at ConcatStream.emit (node:events:526:35)
    at finishMaybe (/Users/prithpalsooriya/Desktop/projects/metamask/metamask-extension/node_modules/readable-stream/lib/_stream_writable.js:630:14)
    at endWritable (/Users/prithpalsooriya/Desktop/projects/metamask/metamask-extension/node_modules/readable-stream/lib/_stream_writable.js:638:3)
    at Writable.end (/Users/prithpalsooriya/Desktop/projects/metamask/metamask-extension/node_modules/readable-stream/lib/_stream_writable.js:594:41)
    at DuplexWrapper.<anonymous> (/Users/prithpalsooriya/Desktop/projects/metamask/metamask-extension/node_modules/duplexer2/index.js:29:14)
    at Object.onceWrapper (node:events:628:28)
    at DuplexWrapper.emit (node:events:526:35)
    at finishMaybe (/Users/prithpalsooriya/Desktop/projects/metamask/metamask-extension/node_modules/readable-stream/lib/_stream_writable.js:630:14)
    at endWritable (/Users/prithpalsooriya/Desktop/projects/metamask/metamask-extension/node_modules/readable-stream/lib/_stream_writable.js:638:3)
    at Writable.end (/Users/prithpalsooriya/Desktop/projects/metamask/metamask-extension/node_modules/readable-stream/lib/_stream_writable.js:594:41)
    at BabelifyStream.onend (node:internal/streams/readable:924:10)
    at Object.onceWrapper (node:events:628:28)
    at BabelifyStream.emit (node:events:514:28)
    at endReadableNT (node:internal/streams/readable:1589:12)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
  filename: '/Users/prithpalsooriya/Desktop/projects/metamask/metamask-extension/app/scripts/snaps/preinstalled-snaps.ts',
  stream: Labeled {
    _readableState: ReadableState {
      objectMode: true,
      highWaterMark: 16,
      buffer: BufferList { length: 0 },
      length: 0,
      pipes: [Labeled],
      pipesCount: 1,
      flowing: true,
      ended: false,
      endEmitted: false,
      reading: true,
      sync: false,
      needReadable: true,
      emittedReadable: false,
      readableListening: false,
      resumeScheduled: false,
      destroyed: false,
      defaultEncoding: 'utf8',
      awaitDrain: 0,
      readingMore: false,
      decoder: null,
      encoding: null
    },
    readable: true,
    _events: [Object: null prototype] {
      end: [Array],
      error: [Function (anonymous)],
      data: [Function: ondata],
      _mutate: [Function]
    },
    _eventsCount: 4,
    _maxListeners: undefined,
    _writableState: WritableState {
      objectMode: true,
      highWaterMark: 16,
      finalCalled: false,
      needDrain: false,
      ending: true,
      ended: true,
      finished: true,
      destroyed: false,
      decodeStrings: true,
      defaultEncoding: 'utf8',
      length: 0,
      writing: false,
      corked: 0,
      sync: false,
      bufferProcessing: false,
      onwrite: [Function (anonymous)],
      writecb: null,
      writelen: 0,
      bufferedRequest: null,
      lastBufferedRequest: null,
      pendingcb: 0,
      prefinished: true,
      errorEmitted: false,
      bufferedRequestCount: 0,
      corkedRequestsFree: [CorkedRequest]
    },
    writable: false,
    allowHalfOpen: true,
    _options: { objectMode: true },
    _wrapOptions: { objectMode: true },
    _streams: [ [Deps], [Transform] ],
    length: 2,
    label: 'deps',
    [Symbol(kCapture)]: false
  }
}
```
</details> 


<details><summary>Summary when using  `import fs from 'node:fs'`</summary>

```
scripts:core:dev:standardEntryPoints: [Error: Can't walk dependency graph: ENOENT: no such file or directory, lstat '/Users/prithpalsooriya/Desktop/projects/metamask/metamask-extension/node:fs'
    required by /Users/prithpalsooriya/Desktop/projects/metamask/metamask-extension/app/scripts/snaps/preinstalled-snaps.ts] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'lstat',
  path: '/Users/prithpalsooriya/Desktop/projects/metamask/metamask-extension/node:fs',
  stream: Labeled {
    _readableState: ReadableState {
      objectMode: true,
      highWaterMark: 16,
      buffer: BufferList { length: 0 },
      length: 0,
      pipes: [Labeled],
      pipesCount: 1,
      flowing: true,
      ended: false,
      endEmitted: false,
      reading: true,
      sync: false,
      needReadable: true,
      emittedReadable: false,
      readableListening: false,
      resumeScheduled: false,
      destroyed: false,
      defaultEncoding: 'utf8',
      awaitDrain: 0,
      readingMore: false,
      decoder: null,
      encoding: null
    },
    readable: true,
    _events: [Object: null prototype] {
      end: [Array],
      error: [Function (anonymous)],
      data: [Function: ondata],
      _mutate: [Function]
    },
    _eventsCount: 4,
    _maxListeners: undefined,
    _writableState: WritableState {
      objectMode: true,
      highWaterMark: 16,
      finalCalled: false,
      needDrain: false,
      ending: true,
      ended: true,
      finished: true,
      destroyed: false,
      decodeStrings: true,
      defaultEncoding: 'utf8',
      length: 0,
      writing: false,
      corked: 0,
      sync: false,
      bufferProcessing: false,
      onwrite: [Function (anonymous)],
      writecb: null,
      writelen: 0,
      bufferedRequest: null,
      lastBufferedRequest: null,
      pendingcb: 0,
      prefinished: true,
      errorEmitted: false,
      bufferedRequestCount: 0,
      corkedRequestsFree: [CorkedRequest]
    },
    writable: false,
    allowHalfOpen: true,
    _options: { objectMode: true },
    _wrapOptions: { objectMode: true },
    _streams: [ [Deps], [Transform] ],
    length: 2,
    label: 'deps',
    [Symbol(kCapture)]: false
  }
}
```
</details>

Reply:
@brad-decker When using `fs` directly like this, it will be replaced by `brfs` at build time with an inlined version of the file. This is the intention for bundling preinstalled Snaps.

Reply:
@FrederikBolding sounds like from your comment that overriding the lint rule in this case is the expected outcome. In which case I'm fine resolving this entry